### PR TITLE
feat: get contact details for rechtspersoon

### PR DESCRIPTION
**Pull request by edgarvonk**
Get contact details for rechtspersoon from Open Klant.

Solves PZ-8437

### CODE REVIEW

File: src/itest/kotlin/nl/info/zac/itest/KlantRestServiceTest.kt [hunk 1]
```diff
@@ -668,18 +671,20 @@ class KlantRestServiceTest : BehaviorSpec({
                     testUser = RAADPLEGER_DOMAIN_TEST_1
```
Comment (edgarvonk) [inline src/itest/kotlin/nl/info/zac/itest/KlantRestServiceTest.kt:671]: ### Review concurrent contact-details enrichment for rechtspersoon KVK lookup

**Risk:** MEDIUM

Replaces the single-service KVK lookup in `readRechtspersoonByKvkNummer` with a concurrent fetch from both KVK and Open Klant, merging email/telephone into the `RestBedrijf` response. Mirrors the exact `runBlocking { withContext(Dispatchers.IO) { async { ... } } }` pattern already established in `readVestiging`.\n\n**Happy path**: Both calls succeed → `emailadres`/`telefoonnummer` populated from Open Klant. **Empty contact path**: `toContactDetails()` returns null fields → response has null email/phone. **Not found path**: KVK returns null → `RechtspersoonNotFoundException` thrown (no contact data retrieved).\n\nThe unit tests cover the happy path and the empty-contact-list edge case. The integration test also validates the enriched response.\n\n**Risks**: If the Open Klant call fails (network error, service down), the entire endpoint fails even though KVK data was successfully retrieved. This is a documented trade-off in the design spec.

**Diagram:**

```mermaid
sequenceDiagram
    participant C as Client
    participant RS as KlantRestService
    participant KVK as KVK Client
    participant OK as Open Klant Client

    C->>RS: GET /rechtspersoon/kvknummer/{kvk}
    RS->>RS: runBlocking + withContext(IO)
    par Concurrent fetch
        RS->>KVK: findRechtspersoonByKvkNummer(kvk)
        RS->>OK: findDigitalAddressesForNonNaturalPerson(kvk)
    end
    KVK-->>RS: rechtspersoon or null
    OK-->>RS: digital addresses list
    alt KVK returns null
        RS-->>C: 404 RechtspersoonNotFoundException
    else KVK found
        RS->>RS: toRestBedrijf() + apply(email, phone)
        alt OK returns empty list
            RS->>RS: email/phone = null (graceful)
        end
        RS-->>C: 200 RestBedrijf (with contact details)
    end
```

File: src/main/kotlin/nl/info/zac/app/klant/KlantRestService.kt [hunk 2]
```diff
@@ -128,16 +134,16 @@ class KlantRestService @Inject constructor(
```
Comment (edgarvonk) [inline src/main/kotlin/nl/info/zac/app/klant/KlantRestService.kt:136]: ### Review undocumented RSIN endpoint behavioral change (kvkNummer no longer nulled)

**Risk:** MEDIUM

The `readRechtspersoonByRsin` endpoint previously explicitly set `kvkNummer = null` via `?.copy(kvkNummer = null)`. This line was removed, meaning the RSIN endpoint now returns the KVK number from the underlying KVK client response instead of null.\n\n**This is an undocumented behavioral change to a public API endpoint.** The design spec explicitly states the RSIN endpoint "stays contact-detail-free by design" as a non-goal — but the kvkNummer change is separate from contact details and is not mentioned.\n\nThe test was updated from `this.kvkNummer shouldBe null` to `this.kvkNummer shouldBe kvkNummer`, confirming the intent. However, consumers who relied on `kvkNummer` being null for RSIN lookups will now see a populated value.\n\n**Recommendation**: Either explicitly document this in the design spec, or add a comment in the code explaining why the `.copy(kvkNummer = null)` was no longer needed. This makes the change visible and intentional rather than a side-effect.

**Diagram:**

```mermaid
flowchart LR
    subgraph Old["Before (RSIN endpoint)"]
        O_RSIN["readRechtspersoonByRsin"]
        O_RESP["RestBedrijf: kvkNummer = null"]
    end
    subgraph New["After (RSIN endpoint)"]
        N_RSIN["readRechtspersoonByRsin"]
        N_RESP["RestBedrijf: kvkNummer = actual value"]
    end
    O_RSIN -->|".copy(kvkNummer = null)"| O_RESP
    N_RSIN -->|"removed nulling"| N_RESP
    O_RESP -.->|"⚠ undocumented change"| N_RESP
```

File: src/main/kotlin/nl/info/zac/app/klant/KlantRestService.kt [hunk 3]
```diff
@@ -128,16 +134,16 @@ class KlantRestService @Inject constructor(
...
      * This endpoint is provided for legacy reasons.
-     * Prefer using the KVK number for retrieving rechtspersonen using [readRechtspersoonByKvkNummer] where possible.
+     * Prefer using the KVK number for retrieving rechtspersonen using [readRechtspersoonByKvkNummer] if possible.
      */
     @GET
     @Path("rechtspersoon/rsin/{rsin}")
     fun readRechtspersoonByRsin(@PathParam("rsin") @Length(min = 9, max = 9) rsin: String): RestBedrijf =
         kvkClientService.findRechtspersoonByRsin(rsin)
             ?.toRestBedrijf()
-            ?.copy(kvkNummer = null)
```
Comment (edgarvonk) [inline src/main/kotlin/nl/info/zac/app/klant/KlantRestService.kt:140]: **Feedback:** Undocumented RSIN response change — kvkNummer no longer nulled<br>
**Severity:** 🔵 Nice to Have

**Agent Opencode:**
Removing `?.copy(kvkNummer = null)` changes the RSIN endpoint response — it now returns the KVK number instead of null. This is undocumented in the design spec, which only mentions the non-goal of keeping the RSIN endpoint "contact-detail-free."

**Recommendation**: Document this in the design spec (e.g., "KVK number is available from the RSIN lookup, no need to null it out") or restore the `.copy` if unintentional. The change is reasonable, but consumers who relied on the null value need visibility.

File: src/main/kotlin/nl/info/zac/app/klant/KlantRestService.kt [hunk 4]
```diff
@@ -146,9 +152,21 @@ class KlantRestService @Inject constructor(
     @GET
     @Path("rechtspersoon/kvknummer/{kvkNummer}")
     fun readRechtspersoonByKvkNummer(@PathParam("kvkNummer") @Length(min = 8, max = 8) kvkNummer: String): RestBedrijf =
-        kvkClientService.findRechtspersoonByKvkNummer(kvkNummer)
-            ?.toRestBedrijf()
-            ?: throw RechtspersoonNotFoundException("Geen rechtspersoon gevonden voor KVK nummer '$kvkNummer'")
+        runBlocking {
+            // run the two client calls concurrently in a coroutine scope,
+            // so we do not need to wait for the first call to complete
```
Comment (edgarvonk) [inline src/main/kotlin/nl/info/zac/app/klant/KlantRestService.kt:159]: **Feedback:** RSIN endpoint kvkNummer no longer nulled<br>
**Severity:** 🔵 Nice to Have

**Agent System:**
readRechtspersoonByRsin removed ?.copy(kvkNummer = null), changing the response shape for RSIN-based lookups. Previously kvkNummer was explicitly null, now it returns the KVK client's value.

**Evidence:** src/main/kotlin/nl/info/zac/app/klant/KlantRestService.kt — removed line `?.copy(kvkNummer = null)` in hunk H2. Test at src/test/.../KlantRestServiceTest.kt hunk H1 changed assertion from `kvkNummer shouldBe null` to `kvkNummer shouldBe kvkNummer`.

**Review by edgarvonk — COMMENTED** [review]
## Merge Confidence: 4.0/5 - Confident

*"Looks good - minor concerns only"*

<details>
<summary>Assessment</summary>

- ✓ Feature is well-scoped: enriches an existing endpoint with contact details from Open Klant
- ✓ Implementation mirrors established readVestiging pattern — concurrent async calls, same error handling
- ✓ Good test coverage: both happy path and empty-contact-list edge case in unit tests; integration test validated
- ✓ Comprehensive spec documentation (openspec) — design decisions, risks, and non-goals are documented
- ⚠ Undocumented RSIN endpoint behavioral change: kvkNummer no longer nulled in response (nice_to_have — should be explicit in design doc)

</details>

---

# Review: Add contact details enrichment for rechtspersoon KVK lookup

Adds concurrent fetch of KVK rechtspersoon data and Open Klant digital addresses, merging email/telephone into the response. Matches the existing readVestiging pattern. Main concern: the RSIN endpoint had `kvkNummer = null` removed without documentation — worth calling out explicitly.

File: src/main/kotlin/nl/info/zac/app/klant/KlantRestService.kt (hunk 2 above)
Comment (edgarvonk) [reply src/main/kotlin/nl/info/zac/app/klant/KlantRestService.kt:136]: this fixes a past mistake. it was never intended not to return a KVK number in case only an RSIN is available.

File: src/main/kotlin/nl/info/zac/app/klant/KlantRestService.kt (hunk 3 above)
Comment (edgarvonk) [reply src/main/kotlin/nl/info/zac/app/klant/KlantRestService.kt:140]: this fixes a past mistake. it was never intended not to return a KVK number in case only an RSIN is available.

File: src/test/kotlin/nl/info/zac/app/klant/KlantRestServiceTest.kt [hunk 5]
```diff
@@ -353,7 +353,7 @@ class KlantRestServiceTest : BehaviorSpec({
                     with(restBedrijf) {
                         this.rsin shouldBe rsin
                         this.naam shouldBe name
-                        this.kvkNummer shouldBe null
+                        this.kvkNummer shouldBe kvkNummer
                         this.vestigingsnummer shouldBe null
                         this.postcode = postcode
                         this.type = type
```
[the remark this answers is not in the document]
Comment (edgarvonk) [reply src/test/kotlin/nl/info/zac/app/klant/KlantRestServiceTest.kt]: good one. fixed

File: src/main/kotlin/nl/info/zac/app/klant/KlantRestService.kt [hunk 6]
```diff
@@ -146,9 +152,21 @@ class KlantRestService @Inject constructor(
...
+            withContext(Dispatchers.IO) {
+                val klantRechtspersoonDigitalAddresses =
+                    async { klantClientService.findDigitalAddressesForNonNaturalPerson(kvkNummer) }
+                val rechtspersoon = async { kvkClientService.findRechtspersoonByKvkNummer(kvkNummer) }
+                klantRechtspersoonDigitalAddresses.await().toContactDetails().let { contactDetails ->
+                    rechtspersoon.await()?.toRestBedrijf()?.apply {
+                        emailadres = contactDetails.emailAddress
+                        telefoonnummer = contactDetails.telephoneNumber
+                    } ?: throw RechtspersoonNotFoundException("Geen rechtspersoon gevonden voor KVK nummer '$kvkNummer'")
+                }
```
[the remark this answers is not in the document]
Comment (edgarvonk) [reply src/main/kotlin/nl/info/zac/app/klant/KlantRestService.kt:171]: nice one. fixed!

**Review by DaxRiool — APPROVED** [review]
Approved. The new added spec files look good. Can't really review those since i would like to see them in action. Code changes look good too!
